### PR TITLE
ENH: Refactor HomeWidget to support custom toolbars

### PR DESCRIPTION
**Pull request by jcfr**
Improves the `HomeWidget` class by introducing convenient functions to managing custom toolbars in the application:

- Adds `toolbarNames` property to dynamically manage toolbar names
- Refactors the `modifyWindowUI` method to create and manage multiple custom toolbars, introducing the following functions:
  - `insertToolBar`: Helper method to insert a new custom toolbar between existing ones.
  - `initializeSettingsToolBar`: Creates the settings toolbar with the gear icon button to toggle visibility of the settings dialog.

### CODE REVIEW

File: {{cookiecutter.project_name}}/Modules/Scripted/Home/Home.py [hunk 1]
```diff
@@ -89,15 +98,38 @@ def setSlicerUIVisible(self, visible: bool):
...
+        appToolBar.setFloatable(False)
         mainToolBar = slicer.util.findChild(slicer.util.mainWindow(), "MainToolBar")
-        self.CustomToolBar = qt.QToolBar("CustomToolBar")
-        self.CustomToolBar.name = "CustomToolBar"
-        slicer.util.mainWindow().insertToolBar(mainToolBar, self.CustomToolBar)
+        slicer.util.mainWindow().insertToolBar(mainToolBar, appToolBar)
+        self._toolbars[appToolBar.name] = appToolBar
+
+    def addToolBar(self, toolBarName: str) -> qt.QToolBar:
+        """Helper method to initialize a new custom toolbar and parent it to the application toolbar"""
```
Comment (sjh26) [inline {{cookiecutter.project_name}}/Modules/Scripted/Home/Home.py:117]: It is not "parented" it is merely inserted before the the named toolbar.  I am not sure that we one to hardcode this, as we maybe be inserting toolbars in various locations for different applications (especially if built in toolbars are used)

Comment (jcfr) [reply {{cookiecutter.project_name}}/Modules/Scripted/Home/Home.py:117]: > appToolBar

This toolbar is intended to include all the toolbars specific to the custom applications, for example:

| |
|--|
| image |
| image |
| image |

Comment (jcfr) [reply {{cookiecutter.project_name}}/Modules/Scripted/Home/Home.py:117]: Supporting inserting toolbar in arbitrary location is indeed sensible. I will revisit the approach.

Comment (sjh26) [reply {{cookiecutter.project_name}}/Modules/Scripted/Home/Home.py:117]: See: https://doc.qt.io/qt-6/qmainwindow.html#insertToolBar

Comment (sjh26) [reply {{cookiecutter.project_name}}/Modules/Scripted/Home/Home.py:117]: Keeping a list of custom toolbars to make show/hide actions easier is a good idea, but the toolbar helper functions just reduce the actual flexibility of toolbar creation

Comment (jcfr) [reply {{cookiecutter.project_name}}/Modules/Scripted/Home/Home.py:117]: > But that's the thing, it does not "include" them. Toolbars don't nest like that. It's just an empty toolbar.

To provide additional context, the idea with the `appToolBar` is support adding  application specific toolbars under only one without having having to use `insertToolBar`.

`insertToolBar` is sensible only if toolbar need to be "injected" within the existing set of Slicer toolbars, which is also a valid use case.

I will then look at supporting the two.

Comment (jcfr) [reply {{cookiecutter.project_name}}/Modules/Scripted/Home/Home.py:117]: After further consideration, I removed the  functions related to `appToolBar` and simplified the API as you suggested.

**Pull request merged**